clients/linux/src/background_work.rs

```diff
@@ -0,0 +1,105 @@
+use crate::messages::{Messenger, Msg};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+pub struct BackgroundWork {
+    pub messenger: Messenger,
+    pub auto_save_state: AutoSaveState,
+    pub auto_sync_state: AutoSyncState,
+}
+
+impl BackgroundWork {
+    pub fn default(m: &Messenger) -> Self {
+        Self {
+            messenger: m.clone(),
+            auto_save_state: AutoSaveState::default(),
+            auto_sync_state: AutoSyncState::default(),
+        }
+    }
+
+    pub fn current_time() -> u128 {
+        SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("Time went backwards")
+            .as_millis()
+    }
+
+    pub fn init_background_work(state: Arc<Mutex<Self>>) {
+        loop {
+            thread::sleep(Duration::from_secs(1));
+            let mut bks = state.lock().unwrap();
+            let m = bks.messenger.clone();
+
+            if bks.auto_save_state.is_active {
+                bks.auto_save_state.auto_save(&m);
+            }
+
+            if bks.auto_sync_state.is_active {
+                let last_change = bks.auto_save_state.last_change;
```

Comment:
Organizing the struct (the data) by splitting out `AutoSyncState` and `AutoSaveState` is good. But when you also delineate the logic like that, you have to do things like this. 

If you were to move the logic into the main struct, you could access these members directly from a `auto_sync` function.

This function would then become a cleaner:

```
if field.is_active {
    do_field();
}

if other_field.is_active {
    do_other_field();
}
```